src/test-unit/query-string.spec.js

```diff
@@ -0,0 +1,114 @@
+        expect(parse('foo=bar')).toEqual({foo: 'bar'});
+    });
+
+    it('parse multiple query string', () => {
+        expect(parse('foo=bar&key=val')).toEqual({
+            foo: 'bar',
+            key: 'val'
+        });
+    });
+
+    it('should handle wthiout value', () => {
+        expect(parse('foo')).toEqual({ foo: null });
+        expect(parse('foo&key=val')).toEqual({
+            key: 'val',
+            foo: null
+        });
+
+        expect(parse('foo&foo')).toEqual({foo: [null, null]});
+        expect(parse('foo=&foo')).toEqual({foo: ['', null]});
+    });
+
+    it('handle `+` correctly', () => {
+        expect(parse('foo+faz=bar+baz++')).toEqual({'foo faz': 'bar baz  '});
+    });
+
+    it('handle multiple of the same key', () => {
+        expect(parse('foo=bar&foo=baz')).toEqual({foo: ['bar', 'baz']});
+    });
+
+    it('handle multiple values and preserve appearence order', () => {
+        expect(parse('a=value&a=')).toEqual({a: ['value', '']});
+        expect(parse('a=&a=value')).toEqual({a: ['', 'value']});
+    });
+
+    it('query strings having indexed arrays', () => {
+        expect(parse('foo[0]=bar&foo[1]=baz')).toEqual({
+            'foo[0]': 'bar',
+            'foo[1]': 'baz'
+        });
+    });
+    
+    it('stringifies', () => {
+        expect(stringify({foo: 'bar'})).toEqual('foo=bar');
+        expect(stringify({foo: 'bar', bar: 'baz'})).toEqual('bar=baz&foo=bar');
+    });
+
+    it('URI encode', () => {
+        expect(stringify({'foo bar': 'baz faz'})).toEqual('foo%20bar=baz%20faz');
+        expect(stringify({'foo bar': 'baz\'faz'})).toEqual('foo%20bar=baz%27faz');
+    });
+    
+    it('handle array value', () => {
+        expect(stringify({
+            abc: 'abc',
+            foo: ['bar', 'baz']
+        })).toEqual('abc=abc&foo=bar&foo=baz');
+    });
+
+    it('handle empty array value', () => {
+        expect(stringify({
+            abc: 'abc',
+            foo: []
+        })).toEqual('abc=abc');
+    });
+    
+    it('should not encode undefined values', () => {
+        expect(stringify({
+            abc: undefined,
+            foo: 'baz'
+        })).toEqual('foo=baz');
+    });
+
+    it('handle null values in array', () => {
+        expect(stringify({
+            foo: null,
+            bar: [null, 'baz']
+        })).toEqual('bar&bar=baz&foo');
+    });
+    
+    it('handle undefined values in array', () => {
+        expect(stringify({
+            foo: null,
+            bar: [undefined, 'baz']
+        })).toEqual('bar=baz&foo');
+    });
+    
+    it('handle undefined and null values in array', () => {
+        expect(stringify({
+            foo: null,
+            bar: [undefined, null, 'baz']
+        })).toEqual('bar&bar=baz&foo');
+    });
+    
+    it('strict encoding', () => {
+        expect(stringify({foo: '\'bar\''})).toEqual('foo=%27bar%27');
+        expect(stringify({foo: ['\'bar\'', '!baz']})).toEqual('foo=%27bar%27&foo=%21baz');
+    });
+});
```

Comment:
Let’s also add tests for stringify with options.filterValues passed to show how it works in different cases